### ISSUE TITLE
Build error with GCC9 on aarch64 and armv7

### ISSUE THREAD

**Issue by ggardet** [ARM-software/armnn#193]
Build error with GCC9 on aarch64 and armv7 (native compilation) with armnn version 19.02.

The error log is:
```
[  421s] In file included from /home/abuild/rpmbuild/BUILD/armnn-19.02/src/backends/aclCommon/test/CreateWorkloadClNeon.hpp:7,
[  421s]                  from /home/abuild/rpmbuild/BUILD/armnn-19.02/src/backends/neon/test/NeonCreateWorkloadTests.cpp:10:
[  421s] /home/abuild/rpmbuild/BUILD/armnn-19.02/src/armnn/test/CreateWorkload.hpp: In instantiation of 'std::unique_ptr<_Tp> {anonymous}::CreateMergerWorkloadTest(armnn::IWorkloadFactory&, armnn::Graph&, const armnn::TensorShape&, unsigned int) [with MergerWorkload = armnn::NeonMergerWorkload; armnn::DataType DataType = armnn::DataType::Float32]':
[  421s] /home/abuild/rpmbuild/BUILD/armnn-19.02/src/backends/neon/test/NeonCreateWorkloadTests.cpp:627:75:   required from 'void CreateWorkloadNeon::NeonCreateMergerWorkloadTest(std::initializer_list<unsigned int>, unsigned int) [with MergerWorkloadType = armnn::NeonMergerWorkload; armnn::DataType DataType = armnn::DataType::Float32]'
[  421s] /home/abuild/rpmbuild/BUILD/armnn-19.02/src/backends/neon/test/NeonCreateWorkloadTests.cpp:641:97:   required from here
[  421s] /home/abuild/rpmbuild/BUILD/armnn-19.02/src/armnn/test/CreateWorkload.hpp:1095:36: error: moving a local object in a return statement prevents copy elision [-Werror=pessimizing-move]
[  421s]  1095 |     return std::move(workloadMerger);
[  421s]       |                                    ^
[  421s] /home/abuild/rpmbuild/BUILD/armnn-19.02/src/armnn/test/CreateWorkload.hpp:1095:36: note: remove 'std::move' call
[  421s] /home/abuild/rpmbuild/BUILD/armnn-19.02/src/armnn/test/CreateWorkload.hpp: In instantiation of 'std::unique_ptr<_Tp> {anonymous}::CreateMergerWorkloadTest(armnn::IWorkloadFactory&, armnn::Graph&, const armnn::TensorShape&, unsigned int) [with MergerWorkload = armnn::NeonMergerWorkload; armnn::DataType DataType = armnn::DataType::QuantisedAsymm8]':
[  422s] /home/abuild/rpmbuild/BUILD/armnn-19.02/src/backends/neon/test/NeonCreateWorkloadTests.cpp:627:75:   required from 'void CreateWorkloadNeon::NeonCreateMergerWorkloadTest(std::initializer_list<unsigned int>, unsigned int) [with MergerWorkloadType = armnn::NeonMergerWorkload; armnn::DataType DataType = armnn::DataType::QuantisedAsymm8]'
[  422s] /home/abuild/rpmbuild/BUILD/armnn-19.02/src/backends/neon/test/NeonCreateWorkloadTests.cpp:656:105:   required from here
[  422s] /home/abuild/rpmbuild/BUILD/armnn-19.02/src/armnn/test/CreateWorkload.hpp:1095:36: error: moving a local object in a return statement prevents copy elision [-Werror=pessimizing-move]
[  422s] /home/abuild/rpmbuild/BUILD/armnn-19.02/src/armnn/test/CreateWorkload.hpp:1095:36: note: remove 'std::move' call
[  422s] cc1plus: all warnings being treated as errors
[  422s] make[2]: *** [src/backends/neon/test/CMakeFiles/armnnNeonBackendUnitTests.dir/build.make:66: src/backends/neon/test/CMakeFiles/armnnNeonBackendUnitTests.dir/NeonCreateWorkloadTests.cpp.o] Error 1
[  422s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/armnn-19.02/build'
[  422s] make[2]: *** Waiting for unfinished jobs....
```

**Comment by MikeJKelly** [comment]
Hi,

the issue is that the -Werror=pessimizing-move compilation flag is creating an error, that's not a required flag when building ArmNN, can you turn it off and try again? 

Best regards,
Mike

**Comment by ggardet** [comment]
> the issue is that the -Werror=pessimizing-move compilation flag is creating an error, that's not a required flag when building ArmNN, can you turn it off and try again?

I would say the compilation flag shows an actual error, so, I would prefer to fix it, instead of trying to ignore it. 
What about removing `std::move` and just return the value?

**Comment by MikeJKelly** [comment]
Good point, I'll create a ticket and add it to our roadmap.

**Comment by ggardet** [comment]
Here is what I tried: 
```
--- armnn-19.02.orig/src/armnn/test/CreateWorkload.hpp	2019-02-28 13:19:24.000000000 +0100
+++ armnn-19.02/src/armnn/test/CreateWorkload.hpp	2019-05-29 17:17:11.912401239 +0200
@@ -1092,7 +1092,7 @@ std::unique_ptr<MergerWorkload> CreateMe
     auto workloadMerger = MakeAndCheckWorkload<MergerWorkload>(*merger, graph, factory);
     BOOST_TEST_CHECKPOINT("created merger workload");
 
-    return std::move(workloadMerger);
+    return workloadMerger;
 }
 
 template <typename PreCompiledWorkload, armnn::DataType dataType>
```

**Comment by ggardet** [comment]
But the error found here: https://github.com/ARM-software/armnn/issues/140#issuecomment-498190719 may be related

**Comment by MikeJKelly** [comment]
Hi,
this build issue should have been fixed by https://review.mlplatform.org/c/ml/armnn/+/1333 I think that it should also have fixed the unit test issue in #140

**Comment by ggardet** [comment]
Fixed in 19.08
